File: 백준/Platinum/9175. Who needs 8 Queens when you can have N？/Who needs 8 Queens when you can have N？.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

let t = Number(input[0]);

function printQueens(arr, N) {
    console.log(N);
    let output = "";
    for (let i = 0; i < N; i += 1) output += " " + arr[i];
    console.log(output);
}

function placeQueens(N, arr) {
    if (N % 6 === 2) {
        let idx = 0;
        for (let i = 2; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 2;
        arr[idx++] = 0;
        for (let i = 7; i <= N; i += 2) arr[idx++] = i - 1;
        if (N !== 5) arr[idx++] = 4;
    } else if (N % 6 === 3) {
        let idx = 0;
        for (let i = 4; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 1;
        for (let i = 5; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 0;
        arr[idx++] = 2;
    } else {
        let idx = 0;
        for (let i = 2; i <= N; i += 2) arr[idx++] = i - 1;
        for (let i = 1; i <= N; i += 2) arr[idx++] = i - 1;
    }
    printQueens(arr, N);
}

for (let i = 1; i <= t; i += 1) {
    let n = Number(input[i]);
    let arr = new Array(n);
    placeQueens(n, arr);
}

File: 백준/Platinum/9175. Who needs 8 Queens when you can have N？/Who needs 8 Queens when you can have N？.c
#include <stdio.h>

void printQueens(int *arr, int N) {
    printf("%d\n", N);
    for (int i = 0; i < N; i++) {
        if (i < N) printf(" ");
        printf("%d", arr[i]);
    }
    printf("\n");
}

void placeQueens(int N, int *arr) {
    if (N % 6 == 2) {
        int idx = 0;
        for (int i = 2; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 2;
        arr[idx++] = 0;
        for (int i = 7; i <= N; i += 2) arr[idx++] = i - 1;
        if (N != 5) arr[idx++] = 4;
    }
    else if (N % 6 == 3) {
        int idx = 0;
        for (int i = 4; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 1;
        for (int i = 5; i <= N; i += 2) arr[idx++] = i - 1;
        arr[idx++] = 0;
        arr[idx++] = 2;
    }
    else {
        int idx = 0;
        for (int i = 2; i <= N; i += 2) arr[idx++] = i - 1;
        for (int i = 1; i <= N; i += 2) arr[idx++] = i - 1;
    }
    printQueens(arr, N);
}

int main() {
    int T;
    scanf("%d", &T);
    while (T--) {
        int n;
        scanf("%d", &n);
        int arr[n];
        placeQueens(n, arr);
    }
    return 0;
}